lib/prometheus/business_collector.rb

```diff
@@ -64,6 +70,28 @@ def collect
       Host.with_policies_or_test_results
           .where(account: client_accounts.select(:account_number)).count
     )
+
+    Host.with_policies_or_test_results.select(
+      "COUNT(id), concat(
+        #{Host::OS_MAJOR_VERSION},
+        '.',
+        #{Host::OS_MINOR_VERSION}
+      ) as version"
+    ).group('version').map(&:attributes).each do |item|
+      @total_systems_by_os.observe(item['count'], version: item['version'])
+    end
+
+    Host.with_policies_or_test_results.where(
+      account: client_accounts.select(:account_number)
+    ).select(
+      "COUNT(id), concat(
+        #{Host::OS_MAJOR_VERSION},
+        '.',
+        #{Host::OS_MINOR_VERSION}
+      ) as version"
+    ).group('version').map(&:attributes).each do |item|
```

Comment:
I wonder how this version concat from JSON blob and group by it would perform in production.

Reply:
Do you have performance concerns?

Reply:
We've introduced metrics in the past that affect performance because of the DB pressure they add. I'm not too concerned about Ruby performance, since it at least should not affect other services like compliance-backend.